semanticdb/symtab/src/main/scala/scala/meta/internal/symtab/GlobalSymbolTable.scala

```diff
@@ -0,0 +1,72 @@
+package scala.meta.internal.symtab
+
+import scala.annotation.tailrec
+import scala.collection.concurrent.TrieMap
+import scala.meta.internal.classpath.ClasspathIndex
+import scala.meta.internal.semanticdb.SymbolInformation
+import scala.meta.io._
+import scala.meta.internal.metacp._
+import scala.meta.internal.scalacp.Scalalib
+import scala.meta.internal.semanticdb.Scala._
+import scala.reflect.NameTransformer
+
+/** A lazy symbol table that returns global symbols on-the-fly from disk. */
+final class GlobalSymbolTable(classpathIndex: ClasspathIndex) extends SymbolTable {
+
+  private val symbolCache = TrieMap.empty[String, SymbolInformation]
```

Comment:
What's the difference between a trie map and a usual map?

Reply:
`TrieMap` is the only concurrent Map in the Scala stdlib I'm aware of, I wanted to make `info().` thread-safe and opted to not use Java collections.